src/jit/codegenxarch.cpp

```diff
@@ -7578,7 +7578,7 @@ void CodeGen::genPutArgStkFieldList(GenTreePutArgStk* putArgStk)
             simdTmpReg = genRegNumFromMask(rsvdRegs & RBM_ALLFLOAT);
             assert(genIsValidFloatReg(simdTmpReg));
         }
-        assert(genCountBits(rsvdRegs) == ((intTmpReg == REG_NA) ? 0 : 1) + ((simdTmpReg == REG_NA) ? 0 : 1));
+        assert(genCountBits(rsvdRegs) == ((unsigned)(intTmpReg == REG_NA) ? 0 : 1) + ((simdTmpReg == REG_NA) ? 0 : 1));
```

Comment:
Assuming I understand what the warning is, it seems cleaner to simply cast the entire RHS to unsigned rather than relying on the subexpression's type trickling out through usual arithmetic conversions.